code/workspaces/client-api/src/auth/permissionChecker.js

```diff
@@ -11,6 +11,22 @@ export const permissionWrapper = (permissionSuffix, ...rest) => permissionCheck(
   ...rest,
 );
 
+export function projectPermissionWrapper(args, permissionSuffix, user, next) {
+  if (!args.projectKey) {
+    logger.error('Auth: permission check: FAILED, projectKey not passed in args');
+    return Promise.reject(new Error(`projectKey not passed, expected suffix: ${permissionSuffix}`));
```

Comment:
I don't think this should be returning a Promise as it isn't an async function. I think you could probably be calling `next(new Error(...)`

Reply:
We agreed we'd look at this under a separate story, as it is actually wrapping the current behaviour.